lib/routes.js

```diff
@@ -117,13 +117,84 @@ module.exports = function(self, options) {
     });
 
     self.route('post', 'batch-force-export', self.sortPageIds, function(req, res) {
-      return self.apos.modules['apostrophe-jobs'].run(req, function(req, id, callback) {
-        return self.forceExport(req, id, req.body.locales, callback);
-      }, {
+      return self.apos.modules['apostrophe-jobs'].runNonBatch(req, run, {
         labels: {
           title: 'Force Export'
         }
       });
+
+      function run(req, reporting, callback) {
+        var results = {};
+        return async.series({
+          addRelated: function(callback) {
+            if (!req.body.related) {
+              return callback(null);
+            }
+            // findRelated is actually quite fast, but fetching 1000 docs
+            // one at a time is not. Fetch batches of 100 docs.
+            // Was also tested at batch size 1 to make sure
+            // multiple batches really work.
+            var batchSize = 100;
+            req.body.ids = self.apos.launder.ids(req.body.ids);
+            var prepend = [];
+            reporting.setTotal(req.body.ids.length);
+            var unfetched = req.body.ids;
+            return fetchBatch();
+            function fetchBatch() {
+              var batch = unfetched.slice(0, batchSize);
+              if (!batch.length) {
+                req.body.ids = prepend.concat(req.body.ids);
+                req.body.ids = _.uniq(req.body.ids);
+                return callback(null);
+              }
+              unfetched = unfetched.slice(batchSize);
+              return self.findDocs(req, { _id: { $in: batch } }).areas(true).joins(true).toArray(function(err, docs) {
+                if (err) {
+                  return callback(err);
+                }
+                return async.eachLimit(docs, 5, function(doc, callback) { 
+                  return self.getRelated([ doc ], function(err, related) {
+                    if (err) {
+                      return callback(err);
+                    }
+                    prepend = prepend.concat(_.pluck(related, '_id')); 
+                    reporting.setTotal(_.uniq(prepend.concat(req.body.ids)).length);
```

Comment:
We keep updating the total as we discover more related docs. So you'll see the total changing even though you don't see the success counter incrementing yet, because we haven't reached the phase of actual export yet.